cmd/temporalite/testdata/temporalite-ui.yaml

```diff
@@ -0,0 +1,5 @@
+tls:
+  caFile: dist/rootCA.pem
+  certFile: dist/client.pem
+  keyFile: dist/client-key.pem
+  serverName: local.dev
```

Comment:
> Allow temporal web/ui to be configured from a yaml file so that the web UI does not break when the API is configured for mTLS.

I'm wondering if it would be possible to be smarter about this specific use case in Temporalite so that users don't have to create a UI specific config file as soon as they enable mTLS.

Would there be a way to detect that mTLS is enabled in Temporal's server config and then automatically configure the UI server connection options appropriately?

Reply:
Possibly, and that might be a thing for another PR. Enabling temporalite for mTLS is something that already requires a config file, so this feels like nothing new.

Reply:
Also, even though temporal-ui is run by this process, do we really want to have it use the same certificate to talk to the frontend? This is what we would have to do if we wanted to reuse the mTLS configuration from the temporal server. While that is technically possible, it feels weird & odd to have to set up mTLS for `internode` and `frontend` and then just have the UI reuse either one. That feels like a decision beyond this PR.

Reply:
👍 not a blocker; we still should be able to pass in a UI config file anyway. One day maybe we'll be able to do away with all ports in Temporalite other than frontend and the ui-server, which will make "internode" communication less weird since it's all a single process already.

I'm not too concerned about reuse though as long as we add an easy readonly mode toggle for the UI.